subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,112 @@
+project('speexdsp',
+        'c',
+        version : '1.2.0',
+        license : 'BSD-3-Clause',
+        default_options : [
+          'c_std=c99',
+          'default_library=both',
+          'warning_level=0'])
+
+# library versioning from speexdsp-1.2.0's configure.ac
+# SPEEXDSP_LT_CURRENT=6, SPEEXDSP_LT_REVISION=1, SPEEXDSP_LT_AGE=5
+lib_current = 6
+lib_revision = 1
+lib_age = 5
+lib_soversion = lib_current - lib_age
+lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
+
+c_args = []
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+# Round up configuration data
+add_project_arguments('-DHAVE_CONFIG_H',
+                      language: 'c')
+cdata = configuration_data()
+
+# Set SIMD config data and flags
+simd_type = 'None'
+if host_cpu_family.startswith('x86')
+  cdata.set('USE_SSE', true)
+  cdata.set('USE_SSE2', true)
+  c_args += ['-mfpmath=sse',
+             '-msse2']
+  simd_type = 'SSE2'
+  # MSVC by-default uses SSE2 instructions (unless overridden)
+  # https://docs.microsoft.com/en-us/cpp/build/reference/arch-x86
+
+elif host_cpu_family == 'arm'
+  cdata.set('USE_NEON', true)
+  c_args += ['-mfpu=neon-vfpv4',
+             '-mfloat-abi=hard',
+             '/arch:VFPv4']
+  simd_type = 'NEON (vfpv4)'
+
+elif host_cpu_family == 'aarch64'
+  cdata.set('USE_NEON', true)
+  c_args += '-mfpu=neon-fp-armv8'
+  simd_type = 'NEON (armv8)'
+endif
```

Comment:
Upstream just uses `-march=armv7-a -mfpu=neon` "if neon is requested", but I'll confess to knowing practically nothing about the various ARM options.

Are these options more fine-tuned for specific unames?

Reply:
`-march=armv7-a` and `march=armv8-a` are both fine.

These represent the generic Armv7.0-A (and 8.0-A) architectures without any incremental architecture extensions. (v7.0-A is basically `-march=x86` and 8.0-A is `-march=x86_64` of the ARM world... as basic as they get)

`-mfpu=neon` is fine, but there's a more generic -`march` extension, the `+simd` flag, that can be tacked onto `-march=armxyz+simd` to request SIMD. I like that it's a bit more generic and its name nicely matches the Meson option.